src/geometries/primitives/cone.jl

```diff
@@ -34,3 +34,21 @@ halfangle(c::Cone) = atan(radius(base(c)), height(c))
 
 Base.isapprox(c₁::Cone, c₂::Cone; atol=atol(lentype(c₁)), kwargs...) =
   isapprox(boundary(c₁), boundary(c₂); atol, kwargs...)
+
+function (cone::Cone)(uφ, ur, uh)
+  T = numtype(lentype(cone))
+  if (uφ < 0 || uφ > 1) || (ur < 0 || ur > 1) || (uh < 0 || uh > 1)
+    throw(DomainError((uφ, ur, uh), "c(φ, r, h) is not defined for φ, r, h outside [0, 1]³."))
+  end
+
+  a = cone.apex
+  R = cone.base.radius
+  b = cone.base.plane.p
+  û = normalize(cone.base.plane.u)
+  v̂ = normalize(cone.base.plane.v)
+  sφ, cφ = sincospi(2uφ)
+  r = R * ur
+  c = b + Vec(r * cφ * û) + Vec(r * sφ * v̂)
+  h̄ = uh * (c - a)
+  a + h̄
```

Comment:
same comment here about internal details. please use the public api of the geometries that are not `Cone`